rust/core-lib/src/editor.rs

```diff
@@ -685,6 +686,48 @@ impl Editor {
         }
     }
 
+    // capitalization behaviour is similar to behaviour in XCode
+    fn capitalize_text(&mut self, view: &mut View) {
+        let mut builder = delta::Builder::new(self.text.len());
+        let mut final_selection = Selection::new();
+
+        for &region in view.sel_regions() {
+            final_selection.add_region(SelRegion::new(region.max(), region.max()));
+            let mut word_cursor = WordCursor::new(&self.text, region.min());
+
+            loop {
+                // capitalize each word in the current selection
+                let (start, end) = word_cursor.select_word();
+
+                if start < end {
+                    let interval = Interval::new_closed_open(start, end);
+                    let word = self.text.slice_to_string(start..end);
+
+                    // first letter is uppercase, remaining letters are lowercase
```

Comment:
I'm trying to think through the Unicode consequences of this. Certainly it seems to be making assumptions that the text is alphabetic, but since this is opt-in, I think maybe that's reasonable. Certainly there are cases this might break a grapheme cluster boundary (like emoji), but in those cases I expect the case-transforms will be no-ops, thus the word will be a no-op as well.